airflow/providers/google/cloud/triggers/vertex_ai.py

```diff
@@ -89,11 +91,97 @@ async def run(self) -> AsyncIterator[TriggerEvent]:
             {
                 "status": status,
                 "message": message,
-                "job": HyperparameterTuningJob.to_dict(job),
+                "job": types.HyperparameterTuningJob.to_dict(job),
             }
         )
 
     def _get_async_hook(self) -> HyperparameterTuningJobAsyncHook:
         return HyperparameterTuningJobAsyncHook(
             gcp_conn_id=self.conn_id, impersonation_chain=self.impersonation_chain
         )
+
+
+class RunPipelineJobTrigger(BaseTrigger):
+    """A trigger that makes async calls to Vertex AI to check the state of a running pipeline job."""
+
```

Comment:
I am not sure about what states can be called successful here. I saw that in some other cases (one is CreateHyperparameterTuningJobTrigger) the Paused state is considered a successful one, but I don't understand why. Can somebody clear this up for me, please?